lib/packwerk/cli.rb

```diff
@@ -57,6 +57,8 @@ def execute_command(args)
         output_result(parse_run(args).check)
       when "update-todo", "update"
         output_result(parse_run(args).update_todo)
+      when "show-offenses", "show"
```

Comment:
I wonder if violations or offense is the right wording here. We should be more consistent with out wording internally. Since we've already gotten things like "offense formatters", this is probably fine.

Reply:
Big plus for this! We call these three things:
- todos (e.g. `update-todo`, `package_todo.yml`
- violations
- offenses

IMO violations and offenses are somewhat aggressive terms and TODO captures the desire to change/fix while being less aggressive and also shorter to write!

I'd be in support of changing all language across the board to `todos`! Some things could be changed right away (e.g. internal implementation variable names), others could probably be changed while not being considered a public API breakage (e.g. output of commands), others would need to be deprecated or have a switchover before a major version change (e.g. CLI commands, offenses formatter, the word violation in package_todo files).

What do you all think? (We could start a discussion too.)

Reply:
Not sure I agree with using the word TODO.
To me, the word TODO in Packwerk has the same meaning as in Rubocop's `.rubocop_todo.yml`, which is there to disable _offenses_ "to be fixed later":

> The generated file `.rubocop_todo.yml` contains configuration to disable cops that currently detect an offense in the code by changing the configuration for the cop, excluding the offending files, or disabling the cop altogether once a file count limit has been reached. ([source](https://docs.rubocop.org/rubocop/configuration.html#:~:text=The%20generated%20file%20.rubocop_todo.yml%20contains%20configuration%20to%20disable%20cops%20that%20currently%20detect%20an%20offense%20in%20the%20code%20by%20changing%20the%20configuration%20for%20the%20cop%2C%20excluding%20the%20offending%20files%2C%20or%20disabling%20the%20cop%20altogether%20once%20a%20file%20count%20limit%20has%20been%20reached.))

I think we could use the same word for _violations_ and _offenses_, but would not conflate the concept with _todo_.
_Offense_ feels a bit more neutral to me (and I guess it also matches Rubocop's vocabulary). The word is also already used more than twice more than _violation_.

```
ag -i offense | wc -l
     799
ag -i violation | wc -l
     325
```

I have a hunch that there might be a will to give _violation_ and _offense_ slightly different meanings, but 1. I'm not sure I'm right and 2. this might not be necessary, or could be expressed more explicitly:
- _offense_: anything that's causing an error (config error, reference that violates declared dependencies, etc)
- _violation_: an _offense_ that is not recorded as a _todo_ (and that triggers an error on `check`).

Reply:
Interestingly, even though offense is used more often in the source code, violation (at least at Gusto) is used much more often by the client. Not sure if it's the same at Shopify, but here everyone uses the word violation, and it's probably in part because the word `violation` is in `package_todo.yml` files explicitly. Also, the output of runs uses the word violation (e.g. "no stale violations detected"), although the word offenses is used too sometimes.

I can buy the different use cases for the words. Something is an offense for example in packwerk if there's a file that can't be parsed correctly. A violation is an offense, but is also what happens when everything works correctly in packwerk, but there's a "violation" of packwerk directives (e.g. dependency usage, private API usage, etc.). TODOs are recorded violations, perhaps short for "violations TODO."

I'm not sure if all of this distinction is useful to the client using this system though, but also I haven't really heard of many folks confused about these terms besides us so perhaps it's not a problem! I'm just glad we replaced `deprecated_references` with `todo` though – I've found that has resolved a lot of language confusion on our end.